java/src/jmri/NamedBeanPropertyDescriptor.java

```diff
@@ -0,0 +1,51 @@
+package jmri;
+
+/**
+ * Describes metadata about a given property key for a NamedBean. This metadata is used by the
+ * BeanTable actions to display and edit the properties in question.
+ *
+ * Created by bracz on 1/19/18.
+ */
+
+public abstract class NamedBeanPropertyDescriptor {
```

Comment:
This is tricky, because the returning classes would have to come from the jmri.jmrit tree, but the implementations for NamedBeanPropertyDescriptor are in the jmri.jmrix tree. At the moment I'll have to go with the default renderers registered in the model, which are fine for Boolean properties.

Reply:
Why are the implementations for NamedBeanPropertyDescriptors in `jmri.jmrix`? The jmrix package is *only* for external connection support, and you are forcing properties of NamedBeans to *only* be used for that purpose, which is artificially limiting.

Also why would a Swing component *have* to be in the `jmri.jmrit` tree? Please point me to the documentation for that, since [library structure](http://jmri.org/help/en/html/doc/Technical/IntroStructure.shtml) is clear that generic Swing components are not under that package.

Reply:
Swing components don't _have_ to be in jmrit, but jmri.jmrit.beantable.EnablingCheckboxRenderer is there. This is the renderer that's needed for a Boolean property to be displayed in a beantable column.

NamedBeanPropertyDescriptor can be anywhere, including in anonymous classes. The use-case in this PR is to have it in the system-specific manager. That is in jmri.jmrix. There can be no link from jmri.jmrix to jmri.jmrit according to the system diagram. I don't quite see where I force anything.

For example a BooleanPropertyDescriptor moved to the jmri package cannot have a getRenderer() giving back an EnablingCheckboxRenderer because that's in jmrit.